images/virt-artifact/patches/028-hotplug-container-disk.patch

```diff
@@ -0,0 +1,1827 @@
+ 			mntOptions := cmd.Flag("options").Value.String()
++			var (
++				uid = -1
++				gid = -1
++			)
+ 			for _, opt := range strings.Split(mntOptions, ",") {
+ 				opt = strings.TrimSpace(opt)
+-				switch opt {
+-				case "ro":
++				switch {
++				case opt == "ro":
+ 					mntOpts = mntOpts | syscall.MS_RDONLY
+-				case "bind":
++				case opt == "bind":
+ 					mntOpts = mntOpts | syscall.MS_BIND
++				case opt == "remount":
++					mntOpts = mntOpts | syscall.MS_REMOUNT
++				case strings.HasPrefix(opt, "uid="):
++					uidS := strings.TrimPrefix(opt, "uid=")
++					uidI, err := strconv.Atoi(uidS)
++					if err != nil {
++						return fmt.Errorf("failed to parse uid: %w", err)
++					}
++					uid = uidI
++					dataOpts = append(dataOpts, opt)
++				case strings.HasPrefix(opt, "gid="):
++					gidS := strings.TrimPrefix(opt, "gid=")
++					gidI, err := strconv.Atoi(gidS)
++					if err != nil {
++						return fmt.Errorf("failed to parse gid: %w", err)
++					}
++					gid = gidI
++					dataOpts = append(dataOpts, opt)
+ 				default:
+ 					return fmt.Errorf("mount option %s is not supported", opt)
+ 				}
+@@ -168,8 +198,17 @@ func main() {
+ 				return fmt.Errorf("mount target invalid: %v", err)
+ 			}
+ 			defer targetFile.Close()
+-
+-			return syscall.Mount(sourceFile.SafePath(), targetFile.SafePath(), fsType, uintptr(mntOpts), "")
++			if uid >= 0 && gid >= 0 {
++				err = os.Chown(targetFile.SafePath(), uid, gid)
++				if err != nil {
++					return fmt.Errorf("chown target failed: %w", err)
++				}
++			}
++			var data string
++			if len(dataOpts) > 0 {
++				data = strings.Join(dataOpts, ",")
++			}
++			return syscall.Mount(sourceFile.SafePath(), targetFile.SafePath(), fsType, uintptr(mntOpts), data)
+ 		},
+ 	}
+ 	mntCmd.Flags().StringP("options", "o", "", "comma separated list of mount options")
+diff --git a/manifests/generated/kv-resource.yaml b/manifests/generated/kv-resource.yaml
+index 66d1b01dbf..43e36b7195 100644
+--- a/manifests/generated/kv-resource.yaml
++++ b/manifests/generated/kv-resource.yaml
+@@ -3307,9 +3307,6 @@ spec:
+     - jsonPath: .status.phase
+       name: Phase
+       type: string
+-    deprecated: true
+-    deprecationWarning: kubevirt.io/v1alpha3 is now deprecated and will be removed
+-      in a future release.
+     name: v1alpha3
+     schema:
+       openAPIV3Schema:
+diff --git a/manifests/generated/operator-csv.yaml.in b/manifests/generated/operator-csv.yaml.in
+index 400d118024..05ee099c67 100644
+--- a/manifests/generated/operator-csv.yaml.in
++++ b/manifests/generated/operator-csv.yaml.in
+@@ -605,6 +605,13 @@ spec:
+           - '*'
+           verbs:
+           - '*'
++        - apiGroups:
++          - subresources.virtualization.deckhouse.io
++          resources:
++          - virtualmachines/addvolume
++          - virtualmachines/removevolume
++          verbs:
++          - update
+         - apiGroups:
+           - subresources.kubevirt.io
+           resources:
```

Comment:
ClusterServiceVersion is not used. It is for the OpenShift.

Reply:
It is generated automatically. I think, i will remove the generated directory from the patch